src/FluentDriver.php

```diff
@@ -117,6 +127,45 @@ public function getMappers()
     }
 
     /**
+     * Add mappings from an array of folders.
+     *
+     * @param string[] $paths
+     *
+     * @throws MappingException
+     */
+    public function addPaths($paths)
+    {
+        $includedFiles = [];
+        foreach ($paths as $path) {
+            if (!is_dir($path)) {
+                throw MappingException::fileMappingDriversRequireConfiguredDirectoryPath($path);
+            }
+
+            $iterator = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($path), \RecursiveIteratorIterator::LEAVES_ONLY);
+
+            foreach ($iterator as $file) {
+                if ($file->getBasename('.php') == $file->getBasename()) {
+                    continue;
+                }
+
+                $sourceFile = realpath($file->getPathName());
+                require_once $sourceFile;
```

Comment:
This is how [doctrine](http://www.doctrine-project.org/api/common/2.4/source-class-Doctrine.Common.Persistence.Mapping.Driver.StaticPHPDriver.html#82-133) finds classes for the StaticPHPDriver